### PR TITLE
[17.0][MIG] stock_quantity_history_location

### PR DESCRIPTION
**Pull request by AnaisLForgeFlow**
Standard Migration

@ForgeFlow

### CODE REVIEW

File: stock_quantity_history_location/wizards/stock_quantity_history.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+
+    <record id="view_stock_quantity_history_location" model="ir.ui.view">
+        <field name="name">Inventory Report at Date and Location</field>
```
Comment (GuillemCForgeFlow) [inline stock_quantity_history_location/wizards/stock_quantity_history.xml]:
```suggestion
        <field name="name">stock.quantity.history.form - stock_quantity_history_location</field>
```
Perhaps we can take the chance to update the view name to make it more searchable

File: stock_quantity_history_location/models/stock_quant.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            if list_view:
+                arch = list_view.get("arch", "")
+                arch_tree = etree.XML(arch)
+                buttons = arch_tree.xpath('//button[@name="action_inventory_at_date"]')
+                for button in buttons:
+                    button.set("string", "Inventory at Date & Location")
+                new_arch = etree.tostring(arch_tree, encoding="unicode")
+                list_view["arch"] = new_arch
+
+        return res
```
Comment (GuillemCForgeFlow) [inline stock_quantity_history_location/models/stock_quant.py]:
```suggestion
    @api.model
    def get_views(self, views, options=None):
        res = super().get_views(views, options=options)
        user_has_group = self.user_has_groups("stock.group_stock_multi_locations")
        list_view = res.get("views", {}).get("list", {})
        if user_has_group and list_view:
            arch = list_view.get("arch", "")
            arch_tree = etree.XML(arch)
            buttons = arch_tree.xpath('//button[@name="action_inventory_at_date"]')
            for button in buttons:
                button.set("string", "Inventory at Date & Location")
            new_arch = etree.tostring(arch_tree, encoding="unicode")
            list_view["arch"] = new_arch
        return res
```
We can do both checks at the beginning, and then we can remove one indentation, just as a smaller improvement. Well resolved 👍🏿

**Review by GuillemCForgeFlow — CHANGES_REQUESTED** [review]
for the rest, functional and code review LGTM 👍🏿

**Review by CasVissers-360ERP — APPROVED** [review]
Functional review

File: stock_quantity_history_location/models/stock_quant.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+from odoo import api, models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    @api.model
+    def get_views(self, views, options=None):
+        res = super().get_views(views, options=options)
```
Comment (rousseldenis) [inline stock_quantity_history_location/models/stock_quant.py:14]: @AnaisLForgeFlow Maybe a good improvement should be adding a little test calling the view.

**Review by rousseldenis — APPROVED** [review]

File: stock_quantity_history_location/models/stock_quant.py (hunk 3 above)
Comment (rousseldenis) [reply stock_quantity_history_location/models/stock_quant.py:14]: @JordiBForgeFlow @LoisRForgeFlow

Comment (GuillemCForgeFlow) [reply stock_quantity_history_location/models/stock_quant.py:14]: @rousseldenis sure, we can add this for better test coverage

**Review by PieterPaulussen — APPROVED** [review]

**Comment by LoisRForgeFlow** [comment]
@GuillemCForgeFlow can you update your review?

**Review by atchuthan — APPROVED** [review]

File: stock_quantity_history_location/models/stock_quant.py (hunk 3 above)
Comment (GuillemCForgeFlow) [reply stock_quantity_history_location/models/stock_quant.py:14]: @rousseldenis I've just added the test case, I believe it should be ready to merge 🙂

**Pull request merged**